app/src/main/java/com/hover/stax/bounties/BountyViewModel.kt

```diff
@@ -0,0 +1,144 @@
+package com.hover.stax.bounties
+
+import android.app.Application
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import androidx.lifecycle.*
+import androidx.lifecycle.Observer
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import com.hover.sdk.actions.HoverAction
+import com.hover.sdk.api.Hover
+import com.hover.sdk.sims.SimInfo
+import com.hover.stax.channels.Channel
+import com.hover.stax.countries.CountryAdapter
+import com.hover.stax.database.DatabaseRepo
+import com.hover.stax.transactions.StaxTransaction
+import com.hover.stax.utils.Utils.getPackage
+import kotlinx.coroutines.*
+import org.koin.java.KoinJavaComponent.get
+import java.util.*
+
+class BountyViewModel(application: Application) : AndroidViewModel(application) {
+    private val repo = get(DatabaseRepo::class.java)
+    @JvmField
+    var country: String = CountryAdapter.codeRepresentingAllCountries()
+
+    val actions: LiveData<List<HoverAction>>
+    val channels: LiveData<List<Channel>>
+    val transactions: LiveData<List<StaxTransaction>>
+    private val filteredBountyChannels: MutableLiveData<List<Channel>?>
+    private val bountyList = MediatorLiveData<List<Bounty>>()
+    private var sims: MutableLiveData<List<SimInfo>?>? = null
+    val bountyEmailLiveData : MutableLiveData<Map<Int, String?>> = MutableLiveData()
+    lateinit var defferedBountyList : Deferred<MutableList<Bounty>>
+
+
+     fun uploadBountyUser(email:String) {
+        viewModelScope.launch(Dispatchers.IO) {
+            val result = BountyEmailNetworking(getApplication()).uploadBountyUser(email)
+            bountyEmailLiveData.postValue(result)
+        }
+    }
+
+    private fun loadSims() {
+        if (sims == null) {
+            sims = MutableLiveData()
+        }
+        Thread { sims!!.postValue(repo.presentSims) }.start()
```

Comment:
Since you're already using coroutines, you can replace the uses of `Thread` here.

Reply:
Thanks, this is now fixed in recent push